src/Catalog/Helpers/Retry.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace NuGet.Services.Metadata.Catalog.Helpers
+{
+    public class Retry
+    {
+        public static async Task IncrementalAsync(
+            Func<Task> runLogicAsync,
+            Func<Exception, bool> shouldRetryOnException,
+            int maxRetries,
+            TimeSpan initialWaitInterval,
+            TimeSpan waitIncrement)
+        {
+            for (int currentRetry = 0; currentRetry < maxRetries; ++currentRetry)
+            {
+                try
+                {
+                    await runLogicAsync();
+                    return;
+                }
```

Comment:
Duplicate code between the two methods. 
Maybe invoke return `await IncrementalAsync(runLogicAsync, shouldRetryOnException, x=>false, maxRetries, initialWaitInterval, waitIncrement) `

Or `await IncrementalAsync(runLogicAsync, shouldRetryOnException, null, maxRetries, initialWaitInterval, waitIncrement) `and null check in the method below.

Reply:
This will not work. One can't do `result = await runLogicAsync()` if `runLogicAsync` returns just `Task`. Also, I wouldn't even be able to call the overload that accepts `Func<Task<T>>` as the first argument, because `Func<Task>` cannot be cast to `Func<Task<T>>`.